include/libhal/usb.hpp

```diff
@@ -521,13 +522,13 @@ class interface
     /// Defines the starting index value for this interface. For example, if
     /// this usb interface contains two sub interfaces, then the interface
     /// number should be `interface` and the second should be `interface + 1`.
-    u8 interface;
+    std::optional<u8> interface;
 
     /// Defines the starting index value for this interface's strings. For
     /// example, if this usb interface has 3 strings, then the starting ID for
     /// those strings would be this value. String 1 would be this value, then
     /// string 2 would be `value + 1` and string 3 would be `value + 2`.
-    u8 string;
+    std::optional<u8> string;
```

Comment:
Please update the documentation to describe the `std::nullopt` state of the optional and what it means.